docs/api/platform/FileHandle.md

```diff
@@ -67,73 +66,75 @@ This can be overridden by the target providing `mbed::mbed_target_override_conso
 
-Device-type `FileHandle`s such as `UARTSerial` are expected to also support non-blocking operation, which permits the `read` and `write` calls to return immediately when unable to transfer data. See reference of these functions for more information.
+Device-type `FileHandle`s, such as `UARTSerial`, are expected to also support nonblocking operation, which permits the `read` and `write` calls to return immediately when unable to transfer data. Please see the API reference pages of these functions for more information.
 
 For a timed wait for data, or to monitor multiple `FileHandle`s, see [`poll`](poll.html)
 
-
 ### Event-driven I/O
 
-If using non-blocking I/O, you will probably want to be know when to next attempt a `read` or `write` if they indicate no data is available. `FileHandle::sigio` lets you attach a `Callback` which will be called whenever the `FileHandle` becomes readable or writable.
+If using nonblocking I/O, you probably want to know when to next attempt a `read` or `write` if they indicate no data is available. `FileHandle::sigio` lets you attach a `Callback`, which is called whenever the `FileHandle` becomes readable or writable.
 
 Important notes on sigio:
 
-* The sigio may be issued from interrupt context. You cannot portably issue `read` or `write` calls directly from this callback, so you should queue an [`Event`](event.html) or wake a thread to perform the `read` or `write`.
-* The sigio callback is only guaranteed when a `FileHandle` _becomes_ readable or writable. So if you do not fully drain the input, or fully fill the output, no sigio may be generated. This is also important on start-up - don't wait for sigio before attempting to read/write for the first time, but only use it as a "try again" signal after seeing an `EAGAIN` error.
-* Spurious sigios are permitted - you can't assume data will be available after a sigio.
-* Given all the above, use of sigio normally implies use of non-blocking mode, or possibly `poll`. 
+- The sigio may be issued from interrupt context. You cannot portably issue `read` or `write` calls directly from this callback, so you should queue an [`Event`](event.html) or wake a thread to perform the `read` or `write`.
+- The sigio callback is only guaranteed when a `FileHandle` _becomes_ readable or writable. If you do not fully drain the input or fully fill the output, no sigio may be generated. This is also important on start-up - don't wait for sigio before attempting to read or write for the first time, but only use it as a "try again" signal after seeing an `EAGAIN` error.
+- Spurious sigios are permitted - you can't assume data will be available after a sigio.
+- Given all the above, use of sigio normally implies use of nonblocking mode or possibly `poll`. 
 
-Ordinary files do not generate sigio callbacks, because they are always readable and writable, so never become readable or writable.
+Ordinary files do not generate sigio callbacks because they are always readable and writable.
```

Comment:
I thought the last comma clause was necessary to spell out the logic, but maybe it was going too far. Up to you.

* The sigio callback is only guaranteed when a `FileHandle` _becomes_ readable or writable
* Ordinary files are always readable and writable
* Therefore they never _become_ readable or writable
* Therefore they never generate a sigio callback.